posthog/hogql/database/schema/util/session_where_clause_extractor.py

```diff
@@ -0,0 +1,329 @@
+from typing import Optional
+
+from posthog.hogql import ast
+from posthog.hogql.ast import CompareOperationOp, ArithmeticOperationOp
+from posthog.hogql.visitor import clone_expr, CloningVisitor, Visitor
+
+SESSION_BUFFER_DAYS = 3
+
+
+class SessionWhereClauseExtractor(CloningVisitor):
```

Comment:
I don't quite fully understand what this class is _actually_ doing. Could you add a descriptive comment 🙏

Reply:
We also have in `WhereClauseExtractor` in `posthog/hogql/database/schema/event_sessions.py`. How do these two differ? Feels like they may do some very similar things